runtime/nvqir/custatevec/CuStateVecCircuitSimulator.cu

```diff
@@ -549,39 +548,9 @@ public:
       return cudaq::State{{stateDimension}, {}};
 
     std::vector<std::complex<ScalarType>> tmp(stateDimension);
-    // Use custatevec accessor to retrieve the view
-    custatevecAccessorDescriptor_t accessor;
-    const uint32_t nIndexBits = std::log2(stateDimension);
-    // Note: we use MSB bit ordering when reporting the state vector
-    // hence, bit ordering vector = [N-1, N-2, ..., 0]
-    std::vector<int32_t> bitOrdering(nIndexBits);
-    std::iota(std::rbegin(bitOrdering), std::rend(bitOrdering), 0);
-    std::size_t extraWorkspaceSizeInBytes = 0;
-    // create accessor view
-    HANDLE_ERROR(custatevecAccessorCreateView(
-        handle, deviceStateVector, cuStateVecCudaDataType, nIndexBits,
-        &accessor, bitOrdering.data(), bitOrdering.size(),
-        /*maskBitString*/ nullptr, /*maskOrdering*/ nullptr,
-        /*maskLen*/ 0, &extraWorkspaceSizeInBytes));
-    // allocate external workspace if necessary
-    void *extraWorkspace = nullptr;
-    if (extraWorkspaceSizeInBytes > 0)
-      HANDLE_CUDA_ERROR(cudaMalloc(&extraWorkspace, extraWorkspaceSizeInBytes));
-
-    // set external workspace
-    HANDLE_ERROR(custatevecAccessorSetExtraWorkspace(
-        handle, accessor, extraWorkspace, extraWorkspaceSizeInBytes));
-
-    // get all state vector components: [0, stateDimension)
-    HANDLE_ERROR(custatevecAccessorGet(handle, accessor, tmp.data(),
-                                       /*begin*/ 0,
-                                       /*end*/
-                                       stateDimension));
-    // destroy descriptor
-    HANDLE_ERROR(custatevecAccessorDestroy(accessor));
-    // free extra workspace if allocated
-    if (extraWorkspaceSizeInBytes > 0)
-      HANDLE_CUDA_ERROR(cudaFree(extraWorkspace));
+    cudaMemcpy(tmp.data(), deviceStateVector,
+               stateDimension * sizeof(std::complex<ScalarType>),
+               cudaMemcpyDeviceToHost);
```

Comment:
This change looks good. 

Only a minor suggestion to wrap the CUDA call in `HANDLE_CUDA_ERROR` for better error handling.

```suggestion
    HANDLE_CUDA_ERROR(cudaMemcpy(tmp.data(), deviceStateVector,
               stateDimension * sizeof(std::complex<ScalarType>),
               cudaMemcpyDeviceToHost));
```